waspls/src/Wasp/LSP/State.hs

```diff
@@ -0,0 +1,46 @@
+module Wasp.LSP.State
+  ( HandlerM,
+    Severity (..),
+    State,
+    ServerConfig,
+  )
+where
+
+import Control.Monad.Trans.Except (ExceptT)
+import Control.Monad.Trans.State.Strict (StateT)
+import Data.Aeson
+import Data.Aeson.Types (prependFailure, typeMismatch)
+import Data.Default (Default (def))
+import Data.Text (Text)
+import Language.LSP.Server (LspT)
+
+type HandlerM =
```

Comment:
Would it make sense to name this monad `ServerM`?

It seems to me that is how they name it in haskell-lsp:
 - https://github.com/haskell/haskell-language-server/blob/fbcd3a46d128f3741d38acccca1c872b5dec13b5/ghcide/src/Development/IDE/LSP/Server.hs#L39
 - https://github.com/haskell/haskell-language-server/blob/0c5a317560c4ee7257ddf03de117868cb7a056c2/ghcide/src/Development/IDE/LSP/HoverDefinition.hs#L56

I don't yet have the great grasp over the whole LSP though so maybe I am wrong!

Reply:
Interesting, I'm not sure! To me, `HandlerM` feels closer to right because it's the monad that each handler is running under, and the server itself is internal to the LSP package. But I'm probably biased because I chose that name. Interesting the haskell language server went for `ServerM`, though.

Reply:
Since you prefer ServerM and I don't have much of a preference, and with your comment on the diagnosticsHandler which isn't really a handler, I've renamed to ServerM.